app/views/NotificationPrefView/index.js

```diff
@@ -0,0 +1,401 @@
+import RocketChat from '../../lib/rocketchat';
+import EventEmitter from '../../utils/events';
+import { LISTENER } from '../../containers/Toast';
+import log from '../../utils/log';
+
+const SectionTitle = React.memo(({ title }) => <Text style={styles.sectionTitle}>{title}</Text>);
+
+const SectionSeparator = React.memo(() => <View style={styles.sectionSeparatorBorder} />);
+
+const Info = React.memo(({ info }) => <Text style={styles.infoText}>{info}</Text>);
+
+SectionTitle.propTypes = {
+	title: PropTypes.string
+};
+
+Info.propTypes = {
+	info: PropTypes.string
+};
+
+const OPTIONS = {
+	desktopNotifications: [
+		{
+			label: 'Default',
+			value: 'default'
+		},
+		{
+			label: 'All Messages',
+			value: 'all'
+		},
+		{
+			label: 'Mentions',
+			value: 'mentions'
+		},
+		{
+			label: 'Nothing',
+			value: 'nothing'
+		}
+	],
+	audioNotifications: [
+		{
+			label: 'Default',
+			value: 'default'
+		},
+		{
+			label: 'All Messages',
+			value: 'all'
+		},
+		{
+			label: 'Mentions',
+			value: 'mentions'
+		},
+		{
+			label: 'Nothing',
+			value: 'nothing'
+		}
+	],
+	mobilePushNotifications: [
+		{
+			label: 'Default',
+			value: 'default'
+		},
+		{
+			label: 'All Messages',
+			value: 'all'
+		},
+		{
+			label: 'Mentions',
+			value: 'mentions'
+		},
+		{
+			label: 'Nothing',
+			value: 'nothing'
+		}
+	],
+	emailNotifications: [
+		{
+			label: 'Default',
+			value: 'default'
+		},
+		{
+			label: 'All Messages',
+			value: 'all'
+		},
+		{
+			label: 'Mentions',
+			value: 'mentions'
+		},
+		{
+			label: 'Nothing',
+			value: 'nothing'
+		}
+	],
+	desktopNotificationDuration: [
+		{
+			label: 'Default',
+			value: 0
+		},
+		{
+			label: '1 second',
+			value: 1
+		},
+		{
+			label: '2 seconds',
+			value: 2
+		},
+		{
+			label: '3 seconds',
+			value: 3
+		},
+		{
+			label: '4 seconds',
+			value: 4
+		},
+		{
+			label: '5 seconds',
+			value: 5
+		}
+	],
+	audioNotificationValue: [
+		{
+			label: 'None',
+			value: 'none None'
+		},
+		{
+			label: 'Default',
+			value: '0 Default'
+		},
+		{
+			label: 'Beep',
+			value: 'beep Beep'
+		},
+		{
+			label: 'Ding',
+			value: 'ding Ding'
+		},
+		{
+			label: 'Chelle',
+			value: 'chelle Chelle'
+		},
+		{
+			label: 'Droplet',
+			value: 'droplet Droplet'
+		},
+		{
+			label: 'Highbell',
+			value: 'highbell Highbell'
+		},
+		{
+			label: 'Seasons',
+			value: 'seasons Seasons'
+		}
+	]
+
+};
+
+@connect(state => ({
+	server: state.server
+}))
+export default class NotificationPrefView extends React.Component {
+	static navigationOptions = () => ({
+		title: I18n.t('Notification_Preference')
+	})
+
+	static propTypes = {
+		navigation: PropTypes.object
+	}
+
+	constructor(props) {
+		super(props);
+		this.rid = props.navigation.getParam('rid');
+		this.rooms = database.objects('subscriptions').filtered('rid = $0', this.rid);
+		this.room = JSON.parse(JSON.stringify(this.rooms[0] || {}));
+		this.state = {
+			room: JSON.parse(JSON.stringify(this.rooms[0] || {}))
```

Comment:
Duplicated code

Reply:
No I need two different objects if I pass the upper one it just pass the reference I am comparing them below